File: 09.Mass.Extinction.Web/ViewModels/ActivityReport.cs
namespace _09.Mass.Extinction.Web.ViewModels;

public class ActivityReport
{
    public long Id { get; set; }
    public ulong InitiatorId { get; set; }
    public string Initiator { get; set; }
    public long StartTime { get; set; }
    public long EndTime { get; set; }
    public string ReportType { get; set; }
    public string Args { get; set; }
    public string Report { get; set; }
}

File: 09.Mass.Extinction.Discord.Bot/data/entities/ActivityReport.js
const { Model, DataTypes } = require("sequelize");

class ActivityReport extends Model {
    static sequelizeInit() {
        return {
            id: {
                field: "ActivityReportId",
                type: DataTypes.INTEGER,
                primaryKey: true,
                autoIncrement: true
            },
            initiator: {
                field: "Initiator",
                type: DataTypes.BIGINT
            },
            startTime: {
                field: "StartTime",
                type: DataTypes.DATE
            },
            endTime: {
                field: "EndTime",
                type: DataTypes.DATE
            },
            reportType: {
                field: "ReportType",
                type: DataTypes.TEXT
            },
            args: {
                field: "Args",
                type: DataTypes.TEXT
            },
            report: {
                field: "Report",
                type: DataTypes.TEXT
            }
        }
    }
}

module.exports = ActivityReport;